torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -712,7 +712,13 @@ void Reducer::mark_bucket_ready(size_t bucket_index) {
     if (comm_hook_ == nullptr) {
       bucket.work = process_group_->allreduce(tensors);
     } else {
-      GradBucket grad_bucket(tensors);
+      GradBucket grad_bucket(
+          tensors,
+          // Since currently we do not support single-process multiple-device
```

Comment:
Do we need to `TORCH_CHECK` this assumption here or do we do that elsewhere already?

Reply:
`TORCH_CHECK` is already done at the comm hook registration time, i.e., register_comm_hook method. This should be sufficient.